extension-style/src/main/java/com/mapbox/maps/extension/style/projection/generated/Projection.kt

```diff
@@ -0,0 +1,166 @@
+    setProperty(PropertyValue("name", name))
+  }
+
+  /**
+   * This is an Expression representation of "name".
+   *
+   * The name of the projection to be used for rendering the map.
+   */
+  val nameAsExpression: Expression?
+    /**
+     * Get the Name property as an Expression
+     *
+     * @return ProjectionName
+     */
+    get() {
+      getPropertyValue<Expression>("name")?.let {
+        return it
+      }
+      name?.let {
+        return Expression.literal(it.value)
+      }
+      return null
+    }
+  /**
+   * Set the Name property
+   *
+   * @param name value of name as Expression
+   */
+  override fun name(name: Expression) = apply {
+    val propertyValue = PropertyValue("name", name)
+    setProperty(propertyValue)
+  }
+
+  /**
+   * Bind projection to the map controller.
+   *
+   * @param delegate The map controller
+   */
+  override fun bindTo(delegate: StyleInterface) {
+    this.delegate = delegate
+    val projectionParams = HashMap<String, Value>()
+    properties.forEach {
+      projectionParams[it.key] = it.value.value
+    }
+    val expected = delegate.setStyleProjection(Value(projectionParams))
+    expected.error?.let {
+      throw MapboxStyleException("Set projection failed: $it")
+    }
+  }
+
+  private fun setProperty(property: PropertyValue<*>) {
+    properties[property.propertyName] = property
+    updateProperty(property)
+  }
+
+  private fun updateProperty(propertyValue: PropertyValue<*>) {
+    val expected = delegate?.setStyleProjectionProperty(
+      propertyValue.propertyName,
+      propertyValue.value
+    )
+    expected?.error?.let {
+      throw MapboxStyleException("Set projection property failed: $it")
+    }
+  }
+
+  private inline fun <reified T> getPropertyValue(propertyName: String): T? {
+    delegate?.let {
+      return try {
+        it.getStyleProjectionProperty(propertyName).unwrap()
+      } catch (e: RuntimeException) {
+        Log.e(TAG, "Get projection property failed: ${e.message}")
+        Log.e(TAG, it.getStyleProjectionProperty(propertyName).toString())
+        null
+      }
+    }
+    throw MapboxStyleException("Get property $propertyName failed: projection is not added to style yet.")
+  }
+
+  /**
+   * Static variables and methods.
+   */
+  companion object {
+    private const val TAG = "Mbgl-Projection"
+  }
+}
+
+/**
+ * This Interface contains all the functions that will be exposed to Koltin DSL.
+ *
+ * Separated the DSL receiver class to this interface to avoid IDE code suggestion for
+ * property getters.
+ */
+@ProjectionDsl
+interface ProjectionDslReceiver {
+  /**
+   * Set the Name property.
+   *
+   * The name of the projection to be used for rendering the map.
+   *
+   * @param name as ProjectionName
+   */
+  fun name(name: ProjectionName = ProjectionName.MERCATOR): Projection
+
+  /**
+   * Set the Name property
+   *
+   * @param name value of name as Expression
+   */
+  fun name(name: Expression): Projection
+}
```

Comment:
Not in this PR, we might want to get rid of the DslReceiver interfaces in v11 and use Builder pattern instead to avoid breaking change.